PR template added and gitignore improved (#867)
cleanlab/cleanlab: .github/pull_request_template.md

## Summary

> What is the reason for this Pull-Request? If adding new functionality, please comment:
> a code snippet showcasing the end-to-end usage of your new methods, and the exact output
> from this code (print out any returned variables).

## Impact

> What is the scope of your changes and who/where do you think they will affect?

> 

Author: ✅ completed this

> 

Author: also fixed the same for cleanvision.

## Summary

> What is the reason for this Pull-Request? If adding new functionality, please comment:
> a code snippet showcasing the end-to-end usage of your new methods, and the exact output
> from this code (print out any returned variables).

## Impact

> > What is the scope of your changes and who/where do you think they will affect?

Reviewer: Let's make it optional to mention individuals that would be affected by the change.

```suggestion
> 🌐 Areas Affected: Enumerate modules, functions, or documentation impacted by your code.
> 👥 Who’s Affected: Mention who might be impacted (if applicable).
```

> > If your change modifies documentation or a tutorial notebook, please include
> > screenshots showing the rendered outputs for a quicker initial review.

Reviewer: ```suggestion
### Screenshots
> 📸 If your changes modify documentation or a tutorial notebook, please include screenshots or GIFs showing the rendered outputs for a quicker initial review, highlighting the main changes.
```

## Testing

> > What testing have you done and verified? Please link to the one unit test

Reviewer: ```suggestion
> 🔍 Testing Done: Outline what kinds of tests you performed.
> 🔗 Test Case Link: Directly link to the most end-to-end check of your new functionality.
```

> > that is most end-to-end check of your new functionality. It's ok if your
> > unit tests are not yet comprehensive when you first open the PR,
> > we can revisit them later!

Reviewer: ```suggestion
### Unaddressed Cases

> It's ok if your unit tests are not yet
> comprehensive when you first open the PR,
> we can revisit them later!
> ⚠️ Mention any aspects of testing that have *not* been covered, and why.
```

-

> ## Ticket(s) or Conversations

Reviewer: We utilize GitHub issues to track, discuss, and manage ongoing development tasks, rather than using the term "tickets".

```suggestion
## Links to Relevant Issues or Conversations
```

> > What Git or Slack items (Issues, threads, etc) that are specifically related to

Reviewer: ```suggestion
> 🔗 What Git or Slack items (Issues, threads, etc) that are specifically related to
```

> this work? Please link them here.

-

## References

> > Include any extra information that may be helpful to reviewers of your
> > Pull-Request here (e.g. relevant URLs or Wiki pages that could help give
> > background information). If relevant, please include additional code snippets
> > (and their outputs/return values) showing alternative ways to use your newly
> > added methods.

Reviewer: ````suggestion
> 📚 Include any extra information that may be helpful to reviewers of your
> Pull-Request here (e.g. relevant URLs or Wiki pages that could help give
> background information). 
> Share links, docs, or sources that facilitated your changes.
> If relevant, please include additional code snippets
> (and their outputs/return values) showing alternative ways to use your newly added methods.

```python
[additional code snippets and explanations]
```

-
````

> -

Reviewer: ```suggestion
## Reviewer Notes
> 💡 Include any specific points for the reviewer to consider during their review.

```

## Summary

> 🎯 **Purpose**: Describe the objective of your changes in this Pull-Request.
> 📜 **Example Usage**:  If adding new functionality, include a small code snippet that demonstrates the the end-to-end usage of your new methods.

> > 📄 **Output**: If possible, share the exact output from this ccode (print out any returned variables).

Author: I feel this (`output`) should go to another section below, such as testing, but 🤷

Reviewer: This point is absolutely valid. We should keep the usage and output together in the "Summary" section both for clarity and accessibility.

Having usage and output side-by-side in the initial section allows for a quick and easy reference, reducing the need to scroll through various sections to understand the impact of the code. I've removed the separate point about "output" to avoid any confusion and streamlined the section to present the expected format of the code snippet with its output. This approach aims to provide a clear and immediate understanding of what to expect from the changes.

Thank you for bringing this up!
